modules/local/validate_clinical/main.nf

```diff
@@ -0,0 +1,143 @@
+    ls main.py
+    # Check if upstream process was successful by checking meta.status
+    if [ "${meta.status ?: 'pass' }" != "pass" ]; then
+        echo "Upstream process failed (meta.status: ${meta.status ?: 'pass'}), skipping payload generation"
+        GENERATION_EXIT_CODE=1
+        ERROR_DETAILS="Skipped payload generation due to upstream failure"
+        
+        # Create placeholder payload file to satisfy Nextflow output requirements
+        echo '{"error": "payload_generation_failed", "message": "Placeholder file created due to upstream failure"}' > "${prefix}_payload.json"
+    else
+        echo "Upstream process successful, proceeding with payload generation"
+        
+        # Run main.py once and capture both exit code and error output
+        main.py \
+        ${sample_file} \
+        ${specimen_file} \
+        ${experiment_file} \
+        ${read_group_file} \
+        ${workflow_file} \
+        --file_metadata ${analysis.files} \
+        --analysis_metadata ${analysis.analysis} \
+        --relational_mapping ${relational_mapping} \
+        --analysis_types ${analysis_types} \
+        --clinical_url ${params.clinical_url} \
+        --file_manager_url ${params.file_manager_url} \
+        --token ${params.token} \
+        --data_directory ${data_directory} \
+        --study_id ${meta.study} 2>generation_errors.tmp
+
+        GENERATION_EXIT_CODE=\$?
+
+        # Capture error details if generation failed
+        ERROR_DETAILS=""
+        if [ \$GENERATION_EXIT_CODE -ne 0 ]; then
+            # Read all error details from captured stderr
+            if [ -f "generation_errors.tmp" ] && [ -s "generation_errors.tmp" ]; then
+                ERROR_DETAILS=\$(cat "generation_errors.tmp" | tr '\\n' ' | ' | sed 's/ | \$//' || echo "Script execution failed")
+            else
+                ERROR_DETAILS="Script execution failed"
+            fi
+        fi
+    fi
+
+    # Create step-specific status file (meta.id used for grouping in channel)
+    cat <<-END_STATUS > "${meta.id}_${task.process.toLowerCase().replace(':', '_')}_status.yml"
+    process: "${task.process}"
+    status: "\$(if [ \$GENERATION_EXIT_CODE -eq 0 ]; then echo 'SUCCESS'; else echo 'FAILED'; fi)"
+    exit_code: \$GENERATION_EXIT_CODE
+    timestamp: "\$(date -Iseconds)"
+    details:
+        analysis_id: "${meta.id}" 
+        sample_meta : "${clinical.sample}"
+        specimen_meta :  "${clinical.specimen}"
+        experiment_meta : "${clinical.experiment}"
+        read_group_meta : "${clinical.read_group}"
+        exit_on_error_enabled: "${exit_on_error_str}"
+    END_STATUS
+    
+    # Add error message to status file if generation failed
+    if [ \$GENERATION_EXIT_CODE -ne 0 ] && [ -n "\$ERROR_DETAILS" ]; then
+        echo "    error_message:" >> "${meta.id}_${task.process.toLowerCase().replace(':', '_')}_status.yml"
+        cat generation_errors.tmp >> "${meta.id}_${task.process.toLowerCase().replace(':', '_')}_status.yml"
```

Comment:
`generation_errors.tmp` won't be generated when Upstream process failed (meta.status: failed) and skip this process. 
You can retrieve the error info directly from $ERROR_DETAILS. 
See ref: https://github.com/Pan-Canadian-Genome-Library/molecular-data-submission-workflow/blob/integrate_test/modules/local/samtools/quickcheck/main.nf#L69-L74